File: java/acronym/src/main/java/Acronym.java
import java.util.Arrays;
import java.util.Collections;
import java.util.stream.Collectors;

class Acronym {
    String phrase;

    Acronym(String phrase) {
         this.phrase = phrase;
    }

    String get() {
        return Arrays.stream(this.phrase.split("(\\s|-)"))
                    .map(w -> w.substring(0,1).toUpperCase())
                    .collect(Collectors.joining());
    }
}

File: csharp/acronym/Acronym.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class Acronym
{


    public static string Abbreviate(string phrase)
    {
        char[] foo = phrase.Split(new char[] { '-', '_', ' ' }, StringSplitOptions.RemoveEmptyEntries)
                .Select(word => word[0])
                .ToArray();
        return string.Join("", foo).ToUpper();
    }
}